### PR TITLE
Update section about release management

### PR DESCRIPTION
**Pull request by kiryldz**
## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -354,6 +354,38 @@ stacktrace that was saved as a txt file.
 $ ndk-stack -sym obj/arm64-v8a -dump trace.txt
 ```
 
+## Release management
+
+The Mapbox Maps SDK for Android makes stable releases every 6 weeks. Between releases, updates are available as snapshots before the issuance of a beta build in week 2 and a release candidate in week 4, which is promoted to stable in Week 6.
+
+### What is a snapshot?
+A snapshot release is a mechanism by which the latest commits are available as a build. Android provides snapshots as unique builds. Snapshots are available continuously.
```
Comment (yunikkk) [inline DEVELOPING.md:362]: Not sure what `Android provides snapshots as unique builds.` means

Comment (tobrun) [reply DEVELOPING.md:362]: +1, is this about us not supporting date/time driven snapshots?

File: DEVELOPING.md [hunk 2]
```diff
@@ -354,6 +354,38 @@ stacktrace that was saved as a txt file.
...
+A beta release is a pre-release that exposes initial APIs and behaviors of upcoming features for early feedback. This release is made during week 2 of the release cadence.
+
+### What is the RC release?
+The scope of this beta release is finalized during week 4 of the cadence, and the release branch is created. Once the release branch is cut, we normally will not introduce any new APIs, and the RC is then promoted to stable in week 6.
+
+### Why do we release on a scheduled cadence?
+We release on a schedule to provide a level of predictability for our customers. Having a cadence that the team can rely on gives transparency and provides a rhythm that becomes second nature.
+
+### What type of versioning does Maps Mobile use?
+We use the standard semantic version scheme (SEMVER) x.y.z to differentiate between major, minor, or patch releases.
```
Comment (tobrun) [inline DEVELOPING.md:378]: maybe add link to https://semver.org/

File: DEVELOPING.md (hunk 1 above)
Comment (kiryldz) [reply DEVELOPING.md:362]: I don't have that much info, I copied it from our internal doc 😆 Will remove, I guess snapshots are well described in a section below.

File: DEVELOPING.md [hunk 3]
```diff
@@ -354,6 +354,40 @@ stacktrace that was saved as a txt file.
 $ ndk-stack -sym obj/arm64-v8a -dump trace.txt
 ```
 
+## Release management
+
+The Mapbox Maps SDK for Android makes stable releases every 6 weeks. Between releases, updates are available as snapshots before the issuance of a beta build in week 2 and a release candidate in week 4, which is promoted to stable in week 6.
```
Comment (yunikkk) [inline DEVELOPING.md:359]: BTW are the snapshots available after beta/rc are cut?

File: DEVELOPING.md [hunk 4]
```diff
@@ -354,6 +354,40 @@ stacktrace that was saved as a txt file.
...
+A snapshot release is a mechanism by which the latest commits are available as a build. Working with snapshots is described in detail in a section below.
+
+### Why do we provide pre-releases?
+We provide pre-releases during development to enable the community and our customers to provide feedback.
+
+### What is the beta release?
+A beta release is a pre-release that exposes initial APIs and behaviors of upcoming features for early feedback. This release is made during week 2 of the release cadence.
+
+### What is the RC release?
+The scope of this beta release is finalized during week 4 of the cadence, and the release branch is created. Once the release branch is cut, we normally will not introduce any new APIs, and the RC is then promoted to stable in week 6.
```
Comment (yunikkk) [inline DEVELOPING.md:371]: Is it true regarding the branch though?

File: DEVELOPING.md [hunk 5]
```diff
@@ -354,6 +354,40 @@ stacktrace that was saved as a txt file.
...
+
+### Why do we release on a scheduled cadence?
```
Comment (yunikkk) [inline DEVELOPING.md:373]: IMO should be higher, 1 or 2 section of the release-management

File: DEVELOPING.md (hunk 3 above)
Comment (kiryldz) [reply DEVELOPING.md:359]: as discussed offline snapshots indeed do have their own section below and are not actually connected with release pipeline, will remove mentioning them here
cc @mapbox/maps-android

File: DEVELOPING.md (hunk 4 above)
Comment (yunikkk) [reply DEVELOPING.md:371]: I mean - we cut release branch before releasing beta, not rc

Comment (kiryldz) [reply DEVELOPING.md:371]: goot catch, should be moved in section above for the beta release

File: DEVELOPING.md (hunk 5 above)
Comment (kiryldz) [reply DEVELOPING.md:373]: will do

File: DEVELOPING.md [hunk 6]
```diff
@@ -354,6 +354,36 @@ stacktrace that was saved as a txt file.
...
+We release on a schedule to provide a level of predictability for our customers. Having a cadence that the team can rely on gives transparency and provides a rhythm that becomes second nature.
+
+### Why do we provide pre-releases?
+We provide pre-releases during development to enable the community and our customers to provide feedback.
+
+### What is the beta release?
+A beta release is a pre-release that exposes initial APIs, behaviors of upcoming features for early feedback, and the release branch is created. We make this release during week 2 of the release cadence.
+
+### What is the RC release?
+We finalize the scope of the beta release during week 4 of the cadence. We normally will not introduce any new APIs, and the RC is then promoted to stable in week 6.
```
Comment (yunikkk) [inline DEVELOPING.md:370]: `we not introduce new APIs and promote RC to stable`?

File: DEVELOPING.md [hunk 7]
```diff
@@ -354,6 +354,36 @@ stacktrace that was saved as a txt file.
 $ ndk-stack -sym obj/arm64-v8a -dump trace.txt
 ```
 
+## Release management
+
+The Mapbox Maps SDK for Android makes stable releases every 6 weeks. Beta build is available in week 2 and a release candidate in week 4, which is promoted to stable in week 6.
```
Comment (yunikkk) [inline DEVELOPING.md:359]: Add a not that we treat stable release as new major or minor release in semver terms?

File: DEVELOPING.md (hunk 6 above)
Comment (yunikkk) [inline DEVELOPING.md:370]:
```suggestion
We normally will not introduce any new APIs and promote RC to stable in week 6.
```

File: DEVELOPING.md [hunk 8]
```diff
@@ -354,6 +354,36 @@ stacktrace that was saved as a txt file.
...
+
+We release on a schedule to provide a level of predictability for our customers. Having a cadence that the team can rely on gives transparency and provides a rhythm that becomes second nature.
+
+### Why do we provide pre-releases?
+We provide pre-releases during development to enable the community and our customers to provide feedback.
+
+### What is the beta release?
+A beta release is a pre-release that exposes initial APIs, behaviors of upcoming features for early feedback, and the release branch is created. We make this release during week 2 of the release cadence.
```
Comment (yunikkk) [inline DEVELOPING.md:367]:
```suggestion
A beta release is a pre-release that exposes initial APIs, behaviours of upcoming features for early feedback. We make this release during week 2 of the release cadence.
```

File: DEVELOPING.md (hunk 6 above)
Comment (kiryldz) [reply DEVELOPING.md]: do not like such shortened version, will stick to mine, improved it a bit

File: DEVELOPING.md (hunk 7 above)
Comment (kiryldz) [reply DEVELOPING.md:359]: added this to semver section

File: DEVELOPING.md [hunk 9]
```diff
@@ -354,6 +354,36 @@ stacktrace that was saved as a txt file.
...
+And “z” represents patch releases where we make backward-compatible bug fixes.
+
+### Understanding release management in Git terms
+As soon as time has come to prepare new beta release, we cut the new Git branch `v[major].[minor]` from latest `main`. Beta, RC and stable releases are always cut and published from this branch.
+
+Any bug fix / feature _related_ to the `v[major].[minor]` release (when beta has already been released) will always be merged to `main` firstly and then either merged / cherry-picked to the `v[major].[minor]` branch.
+
+All bug fixes / features _not related_ to the `v[major].[minor]` release (when beta has already been released) will be simply merged to `main` and included in the _next_ release.
+
+If there is the necessity to prepare the patch release for `v[major].[minor]`, we release it from associated `v[major].[minor]` branch with correct `android-v[major].[minor].[patch]` tag and _not_ from separate patch branch.
```
Comment (yunikkk) [inline DEVELOPING.md:385]: WDYT about creating separate section about patch releases below?
e.g. 

#### Patch releases

Comment (kiryldz) [reply DEVELOPING.md:385]: 💎

File: DEVELOPING.md [hunk 10]
```diff
@@ -354,6 +354,43 @@ stacktrace that was saved as a txt file.
...
+
+Any bug fix / feature _related_ to the `v[major].[minor]` release (when beta has already been released) will always be merged to `main` firstly and then either merged / cherry-picked to the `v[major].[minor]` branch.
+
+All bug fixes / features _not related_ to the `v[major].[minor]` release (when beta has already been released) will be simply merged to `main` and included in the _next_ release.
+
+#### Preparing patch release
+If there is the necessity to prepare the patch release for `v[major].[minor]`, we release it from associated `v[major].[minor]` branch with correct `android-v[major].[minor].[patch]` tag and _not_ from separate patch branch.
+
+Preparing patch checklist:
+ - cut `v[major].[minor].[patch]` branch from `v[major].[minor]` branch.
```
Comment (pengdev) [inline DEVELOPING.md:389]: does it have to be a branch named `v[major].[minor].[patch]`? As it seems to be a temporary branch and will be deleted after merging to `v[major].[minor]` branch.

**Review by ank27 — APPROVED** [review]

File: DEVELOPING.md (hunk 10 above)
Comment (yunikkk) [reply DEVELOPING.md:389]: +1, I'd rather not specify and name may be artitrary

**Review by yunikkk — APPROVED** [review]

File: DEVELOPING.md (hunk 10 above)
Comment (kiryldz) [reply DEVELOPING.md]: fixed

File: DEVELOPING.md [hunk 11]
```diff
@@ -354,6 +354,43 @@ stacktrace that was saved as a txt file.
...
+
+All bug fixes / features _not related_ to the `v[major].[minor]` release (when beta has already been released) will be simply merged to `main` and included in the _next_ release.
+
+#### Preparing patch release
+If there is the necessity to prepare the patch release for `v[major].[minor]`, we release it from associated `v[major].[minor]` branch with correct `android-v[major].[minor].[patch]` tag and _not_ from separate patch branch.
+
+Preparing patch checklist:
+ - cut patch branch from `v[major].[minor]` branch.
+ - add patch bug fixes / features, updated the changelog.
+ - merge `v[major].[minor].[patch]` to `v[major].[minor]`, delete patch branch, and start patch release process by creating `android-v[major].[minor].[patch]` tag on `v[major].[minor]` branch.
```
Comment (pengdev) [inline DEVELOPING.md:391]:
```suggestion
 - merge the patch branch to `v[major].[minor]`, delete patch branch, and start patch release process by creating `android-v[major].[minor].[patch]` tag on `v[major].[minor]` branch.
```

**Review by yunikkk — APPROVED** [review]

**Pull request merged**